### PR TITLE
Bug fix/clean replication api wal tracking

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Fixes a theoretical problem with https://github.com/arangodb/arangodb/pull/9416. Namely, if a usual asynchronous replication client would be connected to a cluster (e.g. arangosync), the SyncerIDs could have clashed.

Also, changes the internal handling of the `serverId` parameter back to an integer type (as it was in 3.4), and instead added a string parameter `clientInfo` to pass additional info useful for debugging purposes. This parameter is filled for synchronous replication.

- [X] Bug-Fix for *devel-branch*
- [X] Strictly *new functionality*

The bug fix here is already included in the backport of #9416, namely https://github.com/arangodb/arangodb/pull/9466. The new `serverId` parameter is not backported.

This needs yet to be done:

- [x] Backport to 3.5 branch

### Testing & Verification

The bugfix is a conceptional problem that would be hard to test. As this would be very unlikely to occur in practice, I'm of the mind that additional tests are unnecessary.

The debug information was manually tested and does not have an impact on functionality in any case.

### Documentation

The documentation is changed only in the way that the deprecation notice of `serverId` was removed, as we will continue to use it for usual asynchronous replication clients as before.

https://github.com/arangodb/docs/pull/103

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5201/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5218/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5232/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5249/

**Comment by goedderz** [comment]
Last CI run is all green.

File: arangod/Replication/ReplicationClients.cpp [hunk 1]
```diff
@@ -34,12 +34,39 @@
...
+      : syncerId(progress.syncerId),
+        clientId(progress.clientId),
+        clientInfo(progress.clientInfo) {}
+
+  SyncerInfo(SyncerId syncerId, TRI_server_id_t clientId, std::string const& clientInfo)
+      : syncerId(syncerId), clientId(clientId), clientInfo(clientInfo) {}
+
+  SyncerId const syncerId;
+  TRI_server_id_t const clientId;
+  std::string const& clientInfo;
```
Comment (jsteemann) [inline arangod/Replication/ReplicationClients.cpp]: Should this really be a reference?
That makes the SyncerInfo potentially unsafe or hard to use, if the original string becomes invalid and is still referenced here. I guess we can allow copying the string into the SyncerInfo for better safety & convenience.

File: arangod/Replication/Syncer.cpp [hunk 2]
```diff
@@ -403,7 +403,7 @@ SyncerId newSyncerId() {
     return SyncerId{TRI_NewServerSpecificTick()};
   }
 
-  return SyncerId{ServerIdFeature::getId()};
+  return SyncerId{0};
```
Comment (jsteemann) [inline arangod/Replication/Syncer.cpp]: mmh, is that really intentional?
If so, IMHO it mandates a comment.

File: arangod/Replication/ReplicationClients.cpp (hunk 1 above)
Comment (goedderz) [reply arangod/Replication/ReplicationClients.cpp]: Well, I intended it only for formatting log messages, as in
```cpp
  LOG_TOPIC("c26ab", TRACE, Logger::REPLICATION)
      << "removing replication client entry for "
      << SyncerInfo{syncerId, clientId, clientInfo}
```
. But if you see it maybe being used in other cases, I've got no objections to using a string, I'll change it.

File: arangod/Replication/Syncer.cpp (hunk 2 above)
Comment (goedderz) [reply arangod/Replication/Syncer.cpp]: Yes, it's intentional. I've added a comment.

**Comment by jsteemann** [comment]
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/5296/

**Comment by goedderz** [comment]
The Jenkins run failed due to the hotbackup test. I've merged devel back and started a new run:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5307/

**Comment by goedderz** [comment]
Jenkins is green.

**Pull request merged**